content/courses/ada-in-practice/chapters/streams_device_driver_api_flexibility.rst

```diff
@@ -0,0 +1,662 @@
+.. _Ada_In_Practice_Using_Streams_Device_Driver_API_Flexibility:
+
+Using Streams for Device Driver API Flexibility
+================================================
+
+.. include:: ../../global.txt
+
+Motivation
+----------
+
+Software interfaces for hardware devices usually support only primitive data
+types. Communications devices such as serial I/O ports and network adapters are
+good examples. Their device drivers provide an API for sending and receiving
+individual 8-bit or 9-bit numeric quantities. Software clients |mdash| either
```

Comment:
Let's change "numeric quantities." to "numeric quantities, or sequences of these."